tests/test_opds.py

```diff
@@ -1358,14 +1358,43 @@ def test_acquisition_links(self):
         eq_('http://opds-spec.org/acquisition/open-access', open_access.attrib.get("rel"))
 
         loan2_links = annotator.acquisition_links(
-            loan2.license_pool, loan2, None, None, feed, loan2.license_pool.identifier)
+            loan2.license_pool, loan2, None, None, feed,
+            loan2.license_pool.identifier
+        )
         # Fulfill and revoke.
         [revoke, fulfill] = sorted(loan2_links, key=lambda x: x.attrib.get("rel"))
         assert 'revoke_loan_or_hold' in revoke.attrib.get("href")
         eq_('http://librarysimplified.org/terms/rel/revoke', revoke.attrib.get("rel"))
         assert "fulfill" in fulfill.attrib.get("href")
         eq_('http://opds-spec.org/acquisition', fulfill.attrib.get("rel"))
 
+        # If a book is ready to be fulfilled, but the library has
+        # hidden all of its available content types, the fulfill link does
+        # not show up -- only the revoke link.
+        hidden = self._default_library.setting(
+            Configuration.HIDDEN_CONTENT_TYPES
+        )
+        available_types = [
+            lpdm.delivery_mechanism.content_type
+            for lpdm in loan2.license_pool.delivery_mechanisms
+        ]
+        hidden.value = json.dumps(available_types)
+
+        # The list of hidden content types is stored in the Annotator
```

Comment:
Is this list of hidden content per library and similar to the `hidden` value above?

Reply:
Yes, the constructor grabs the same ConfigurationSetting I modified above and uses its value to configure the Annotator.